### DISCUSSION TITLE
Find sender Address in a transaction

### DISCUSSION THREAD

**Discussion by Blackcode** [dkackman/chia-dotnet#29] [Q&A] [answered]
ToAddress and ToPuzzleHash is the receiving address but how can I find the sender address in a transaction so I can return XCH to him?

**Comment by dkackman** [comment]
The sender is the address of the parent coin. I believe that is the ParentCointInfo on any of the coins in the SpendBundle. However the best place to ask is on the chia Keybase dev channel.

**Comment by Blackcode** [comment, marked as answer]
Thanks man, found the answer last night 👍.
In a transaction you can find it in Transaction -> Additions -> ParentCoinInfo.

The only thing is that is a Puzzle instead of an Address. 
I had to adapt a Bech32 to a Bech32m library to convert.

Thanks

**Comment by dkackman** [comment]
Excellent.  If it helps Ive got some bech32m code over here https://github.com/dkackman/rchia/tree/main/src/rchia/Bech32

**Comment by dkackman** [comment]
Moved that code into chia.dotnet since it's pretty handy to have around: https://github.com/dkackman/chia-dotnet/releases/tag/v0.8.0
